### PR TITLE
Proto conversion traits

### PR DESCRIPTION
**Pull request by peterschwarz**
This PR takes the various proto<->native traits from transact, updates the code based on the latest Rust (1.42 as of this PR), as well as makes some modifications to support more auto trait implementations. 

The largest change is `To-/FromBytes` which adds a generic parameter, `ViaProtocol`.  This allows for the auto-trait implementation for traits where the `ViaProtocol` is a `protobuf::Message`.

Additionally, `IntoNative/-Proto` have also been turned into auto trait implementations, vs default function implementations.

With these updates, consumers of the api only need to implement `FromNative` and `FromProto` for their structs.

A simplification for users of these traits is to also re-export the traits and error via a prelude module.  Use can now do:

```
use splinter::protos::prelude::*;
```

and gain all the benefits of the traits.

### CODE REVIEW

File: libsplinter/src/protos/mod.rs [hunk 1]
```diff
@@ -12,4 +13,110 @@
...
+where
+    P: FromNative<N>,
+{
+    fn into_proto(self) -> Result<P, ProtoConversionError> {
+        FromNative::from_native(self)
+    }
+}
+
+pub mod prelude {
+    //! Allows for he convenient requirement of the proto convertions traits.
```
Comment (agunde406) [inline libsplinter/src/protos/mod.rs:111]: typo "he" -> "the"

File: libsplinter/src/protos/mod.rs [hunk 2]
```diff
@@ -81,20 +81,28 @@ where
     }
 }
 
-pub trait IntoNative<T>: Sized
+pub trait IntoNative<T>: Sized {
```
Comment (vaporos) [inline libsplinter/src/protos/mod.rs:84]: Typo in the commit message "implemetnations"

File: libsplinter/src/protos/mod.rs (hunk 1 above)
Comment (peterschwarz) [reply libsplinter/src/protos/mod.rs:111]: Fixed

File: libsplinter/src/protos/mod.rs (hunk 2 above)
Comment (peterschwarz) [reply libsplinter/src/protos/mod.rs:84]: Updated

**Comment by peterschwarz** [comment]
Added the commit info to the first commit.

**Pull request merged**